File: Sid's Levels/InterviewBit/Maths/Largest Prime Factor.cpp
class Solution{
public: 
    //OM GAN GANAPATHAYE NAMO NAMAH 
    //JAI SHRI RAM 
    //JAI BAJRANGBALI 
    //AMME NARAYANA, DEVI NARAYANA, LAKSHMI NARAYANA, BHADRE NARAYANA
    long long int largestPrimeFactor(int n){
        // code here
        vector<int> spf(n+1, -1);
        for(int i = 0; i <= n; i++)
        spf[i] = i;
        for(int i = 2; i*i <= n; i++)
        {
            if(spf[i] == i)
            {
                for(int j = i*i; j <= n; j+=i)
                {
                    if(spf[j] == j)
                    spf[j] = i;
                }
            }
        }
        int x = n;
        int maxEle = INT_MIN;
        while(x != 1)
        {
            maxEle = max(spf[x], maxEle);
            x = x/spf[x];
        }
        return maxEle;
    }
};

File: DSA Revision/Sid/Mathematics/Largest Prime Factor.java
class Solution{
    //OM GAN GANAPATHAYE NAMO NAMAH 
    //JAI SHRI RAM 
    //JAI BAJRANGBALI 
    //AMME NARAYANA, DEVI NARAYANA, LASKHMI NARAYANA, BHADRE NARAYANA
    static long largestPrimeFactor(int n) {
        // code here
        int lpf = 1;
        while(n%2 == 0)
        {
            lpf = 2;
            n = n/2;
        }
        for(int i = 3; i*i <= n; i++)
        {
            while(n%i == 0)
            {
                lpf = i;
                n = n/i;
            }
        }
        if(n > 1)
        lpf = n;
        return lpf;
    }
}
